### DISCUSSION TITLE
How config check works

### DISCUSSION THREAD

**Discussion by joseacl** [kube-logging/logging-operator#1334] [Q&A] [answered]
Hi!
I wanted to confirm with you how the operator handles configuration checks, because I think I was confused about it.

At the first glance I was thinking that every time that a new Flow/ClusterFlow, Output/ClusterOutput is created/modified, the logging operator runs a pod with a fluentd configuration in dry-run mode, and if that fails the actual configuration is NOT modified. But I've done several tests with that case and even if the dry-run mode fails, this configuration ends up in breaking the original fluentd.
I've also checked the code, and I think is how is it implemented, isn't it?

**Comment by pepov** [comment]
No, of course it shouldn't propage the broken config, can you give a concrete example?

**Comment by pepov** [comment, marked as answer]
You are right, during a change in the cleanup we lost the result error check, here is the fix:
https://github.com/kube-logging/logging-operator/pull/1335

**Reply by joseacl** [reply]
Thanks a lot, @pepov 👍

**Comment by pepov** [comment]
Note: it only affected fluentd, but not the syslog-ng based aggergator
